cerbero/packages/linux.py

```diff
@@ -104,21 +105,24 @@ def build(self, output_dir, tarname, tmpdir, packagedir, srcdir):
         pass
 
     def pack_deps(self, tmpdir):
+        paths = []
         for p in self.store.get_package_deps(self.package.name):
             stamp_path = os.path.join(tmpdir, p.name + '-stamp')
             if os.path.exists(stamp_path):
-                # already built, skipping
+                # Already built, skipping
                 continue
-            p.pre_package()
             m.action(_('Packing dependency %s for package %s') %
                      (p.name, self.package.name))
+            p.pre_package()
             packager = self.__class__(self.config, p, self.store)
+            dep_paths = []
             try:
-                paths = packager.pack(self.output_dir, self.devel, self.force, True, self.split, True, tmpdir)
+                dep_paths = packager.pack(self.output_dir, self.devel, self.force, True, self.split, True, tmpdir)
             except EmptyPackageError:
                 self._empty_packages.append(p)
-                paths = []
-            p.post_package(paths, self.output_dir)
+            finally:
+                paths += p.post_package(dep_paths, self.output_dir)
```

Comment:
post_package does not return a new list of packages to be appended, post_package takes the current list of packages, modifies it and return the new list of packages

Reply:
post_package takes dep_paths as argument, the paths variable is accumulating files for all deps.